File: CIVITASRUBY/lib/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  
  require_relative 'casilla'
  require_relative 'Sorpresa'
  class Casilla_sorpresa < Casilla
    
    def initialize(carcel,mazo, nombre)
      @mazo = mazo
      @sorpresa = nil
      super(nombre,carcel)
    end
    
     def recibe_jugador(actual,todos)
        if jugador_correcto(actual,todos) == true

          @sorpresa = @mazo.siguiente()
          informe(actual,todos)
          @sorpresa.aplicar_a_jugador(actual, todos)
      end
     end
     
     def to_s
      result = "Casilla: \nnombre= #{@nombre} \n tipo= #{@tipo} "
       if  @sorpresa != nil
         result +="\n"
         result += @sorpresa.to_s
       end
      result +="\n"
      
      result
     end
  end
end

File: CIVITAS/src/civitas/CasillaSorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class CasillaSorpresa extends Casilla {
     
    private static MazoSorpresas mazo;
    private  Sorpresa sorpresa;
    
    CasillaSorpresa(MazoSorpresas mazo, String nombre){
        super(nombre);
        CasillaSorpresa.mazo = mazo;
        
    }

    public Sorpresa getSorpresa() {
        return sorpresa;
    }
    
    @Override
    public void updateMazo(MazoSorpresas mazo){
        if(!(CasillaSorpresa.mazo == null))
            CasillaSorpresa.mazo = mazo;
    }

    
    @Override
    public void recibeJugador(int iactual , ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(iactual, todos)){
           sorpresa = CasillaSorpresa.mazo.siguiente();
           this.informe(iactual, todos);
           sorpresa.aplicarAJugador(iactual, todos);
        }
            
    }
    
    
      
    @Override
     public String toString() {
        String result = "\nCasilla{\n" + "nombre=" + this.getNombre();
      
        if(this.sorpresa != null){
            result += this.sorpresa.toString();
        }
        result += "\n}";
        return result;
    }


}
